rqt_py_common/src/rqt_py_common/message_helpers.py

```diff
@@ -0,0 +1,200 @@
+    try:
+        content, _ = get_resource('rosidl_interfaces', package_name)
+    except LookupError:
+        return []
+    interface_names = content.splitlines()
+    # TODO(dirk-thomas) this logic should come from a rosidl related package
+    # Only return services in srv folder
+    all_srvs = [
+        n[4:-4] for n in interface_names if n.startswith('srv/') and n.endswith('.srv')]
+    all_srvs.extend(
+        [n[:-4] for n in interface_names if not n.startswith('srv/') and n.endswith('.srv')])
+    return all_srvs
+
+
+def get_all_message_types():
+    all_message_types = {}
+    for package_name in get_resources('rosidl_interfaces').keys():
+        message_types = get_message_types(package_name)
+        if message_types:
+            all_message_types[package_name] = message_types
+    return all_message_types
+
+
+def get_message_types(package_name):
+    if not has_resource('packages', package_name):
+        raise LookupError('Unknown package name')
+    try:
+        content, _ = get_resource('rosidl_interfaces', package_name)
+    except LookupError:
+        return []
+    interface_names = content.splitlines()
+    # TODO(dirk-thomas) this logic should come from a rosidl related package
+    # Only return messages in msg folder
+    all_msgs = [
+        n[4:-4] for n in interface_names if n.startswith('msg/') and n.endswith('.msg')]
+    all_msgs.extend(
+        [n[:-4] for n in interface_names if not n.startswith('msg/') and n.endswith('.msg')])
+    return all_msgs
+
+
+def _get_message_service_class_helper(message_type, mode, logger):
+    if not (mode == MSG_MODE or mode == SRV_MODE):
+        logger.warn('invalid mode {}'.format(mode))
+        return None
+
+    message_info = message_type.split('/')
+    if len(message_info) == 2:
+        package = message_info[0]
+        base_type = message_info[1]
+    elif len(message_info) == 1:
+        package = 'std_' + mode + 's'
+        base_type = message_info[0]
+    else:
+        logger.error(
+            'Malformed message_type: {}'.format(message_type))
+        return None
+
+    _, resource_path = get_resource('rosidl_interfaces', package)
+    python_pkg = class_val = None
+    try:
+        # import the package
+        python_pkg = __import__('%s.%s' % (package, mode))
+    except ImportError:
+        logger.error('Failed to import class: {} as {}.{}'.format(message_type, package, mode))
+
+    if python_pkg:
+        try:
+            class_val = getattr(getattr(python_pkg, mode), base_type)
+        except AttributeError:
+            if len(base_type):
+                # Try the same thing but with the first letter capitalized
+                base_type = ''.join([base_type[0].upper(), base_type[1:]])
+
+        if not class_val:
+            try:
+                class_val = getattr(getattr(python_pkg, mode), base_type)
+            except AttributeError:
+                logger.error('Failed to load class: {}'.format(message_type))
```

Comment:
Include exception in error message. It won't be clear why this failed in the error message otherwise

Reply:
I'm not sure I agree. I think that if you call get_message_class("std_msgs/not_a_msg") it should tell you that it failed to load and return `None`. The full traceback would be spammy